internal/sqlsmith: TestGenerateParse failed with "unknown survival goal: 0"

internal/sqlsmith.TestGenerateParse [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4011278&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4011278&tab=artifacts#/) on master @ [5ac733bb4927020bc1c52da24b2591742fde8e1f](https://github.com/cockroachdb/cockroach/commits/5ac733bb4927020bc1c52da24b2591742fde8e1f):


```
			SELECT
				*
			FROM
				(VALUES ((-5812699291702972601):::INT8), (2854654142395412186:::INT8), (1947366985986041646:::INT8))
					AS tab_137 (col_244)
		)
SELECT
	(-3653132018074211088):::INT8 AS col_245
FROM
	with_53 AS cte_ref_20
ORDER BY
	cte_ref_20.col_244, cte_ref_20.col_244 ASC, cte_ref_20.col_244, cte_ref_20.col_244;

    panic.go:1047: unknown survival goal: 0
        goroutine 9710 [running]:
        runtime/debug.Stack()
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x72
        github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func1()
        	/go/src/github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:110 +0x94
        panic({0x6942b40, 0xc0005bb330})
        	/usr/local/go/src/runtime/panic.go:1047 +0x266
        github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Stop(0xc002f9c240, {0xa6ae6d8, 0xc0001c2008})
        	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:548 +0x49a
        panic({0x6942b40, 0xc0005bb330})
        	/usr/local/go/src/runtime/panic.go:1047 +0x266
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*SurvivalGoal).Format(0xc0005f9108, 0xc005210230)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/survival_goal.go:39 +0xea
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*AlterDatabaseSurvivalGoal).Format(0xc0005f90f8, 0xc005210230)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/alter_database.go:96 +0x86
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*FmtCtx).formatNodeOrHideConstants(0xc005210230, {0xa61e760, 0xc0005f90f8})
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/hide_constants.go:46 +0x1e4
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*FmtCtx).FormatNode(0xc005210230, {0xa61e760, 0xc0005f90f8})
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/format.go:444 +0x12d
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.AsStringWithFlags({0xa61e760, 0xc0005f90f8}, 0x48040, {0x0, 0x0, 0x0})
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/format.go:582 +0x8e
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*PrettyCfg).docAsString(0x18, {0xa61e760, 0xc0005f90f8})
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/pretty.go:160 +0x45
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*PrettyCfg).Doc(0xc002b0ef60, {0xa61e760, 0xc0005f90f8})
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/pretty.go:155 +0x91
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*PrettyCfg).Pretty(0xd6a9380, {0xa61e760, 0xc0005f90f8})
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/pretty.go:144 +0x45
        github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.(*Smither).Generate(0xa625c20)
        	/go/src/github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/sqlsmith.go:166 +0x7b
        github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.TestGenerateParse(0xc000e68d00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/sqlsmith_test.go:182 +0x9db
        testing.tRunner(0xc000e68d00, 0x7028900)
        	/usr/local/go/src/testing/testing.go:1259 +0x230
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1306 +0x727
--- FAIL: TestGenerateParse (2.99s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-race -parallel=4
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestGenerateParse.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/internal/sqlsmith/alter.go
+++ b/pkg/internal/sqlsmith/alter.go
@@ -462,8 +462,13 @@ func makeAlterDatabaseDropRegion(s *Smither) (tree.Statement, bool) {
 }
 
 func makeAlterSurvivalGoal(s *Smither) (tree.Statement, bool) {
-	const numSurvivalGoals = 3
-	randInt := rand.Intn(numSurvivalGoals)
+	const numSurvivalGoals = 2
+	// Only {1, 2} are valid values for SurvivalGoal.
+	// 1. SurvivalGoalRegionFailure
+	// 2. SurvivalGoalZoneFailure
+	//
+	// 0. SurvivalGoalDefault is not valid as a value to alter to.
+	randInt := rand.Intn(numSurvivalGoals) + 1
 	survivalGoal := tree.SurvivalGoal(randInt)
 
 	ast := &tree.AlterDatabaseSurvivalGoal{

